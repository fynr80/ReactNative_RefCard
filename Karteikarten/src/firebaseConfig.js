
import firebase from 'firebase/compat/app';

const firebaseConfig = {
    apiKey: "AIzaSyBksswFLr1YB6en-sRf8mqSFle7bhAzSgU",
    authDomain: "swtp-812df.firebaseapp.com",
    databaseURL: "https://swtp-812df-default-rtdb.firebaseio.com",
    projectId: "swtp-812df",
    storageBucket: "swtp-812df.appspot.com",
    messagingSenderId: "44236558384",
    appId: "1:44236558384:web:2f15f6b5fb190634b71df5",
    measurementId: "G-W33L0G7D18"
};
firebase.initializeApp(firebaseConfig);
export {firebase};
