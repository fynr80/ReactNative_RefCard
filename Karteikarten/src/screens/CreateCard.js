/**
 * @file CardDescriptionScreen for RefCard
 * @author Ibrahim Eraslan <ibrahim.eraslan@mni.thm.de>
 */

import React, {useState} from 'react';
import {
    StyleSheet,
    Text,
    TextInput,
    TouchableOpacity,
    View
} from 'react-native';

import {SafeAreaView} from 'react-native-safe-area-context';
import {Button} from 'react-native-paper'
import firebase from "firebase/compat";
//import {LinearGradient} from "expo-linear-gradient";
import Icon from "react-native-vector-icons/Ionicons";
import {useNavigation} from "@react-navigation/native";

import stylejs from "../style";


const CreateCard = (props) => {
    const [answer, setAnswer] = useState("")
    const [question, setQuestion] = useState("")

    const navigation = useNavigation()
    const id = props.route.params.id
    const categoryColor = props.route.params.categoryColor;
    const catId = props.route.params.catId
    const title = props.route.params.titel

    /**
     * @description Adds the question to the current set
     * @returns {Promise<void>}
     */
    const createQuestion = async () => {
        await firebase.firestore()
            .collection('category')
            .doc(catId)
            .collection("set")
            .doc(id)
            .collection("card")
            .add({
                question: question,
                answer: answer,
                date: new Date(),
                box: 1
            }).then(() => {
                setQuestion("");
                setAnswer("");
            });
    }

    return (
        <SafeAreaView style={styles.container}>
            {/*Cancel-, Info- and User-Icon*/}
            <SafeAreaView style={[stylejs.headerNew, {width: "100%", alignItems: "flex-start", justifyContent: "center", top: -50}]}>
                <TouchableOpacity
                    onPress={() => {
                        navigation.navigate('Main');
                    }}
                    style={{flex:1}}
                >
                    <Icon name='chevron-back-sharp' size={50} color= '#FD4365' style={{marginTop: -15}}/>
                </TouchableOpacity>
                <Text style={[stylejs.headerText, {textAlign: "center"}]}>CREATE CARDS</Text>
                <Text style={{flex: 1}}></Text>
            </SafeAreaView>
            <Text style={[stylejs.headerText, {textAlign: "center", flex: 1}]}>{title}</Text>
            {/*<LinearGradient
                colors={[categoryColor.left, categoryColor.right]}
                style={styles.inputContainer}
            >*/}
                <View style={styles.inputContainer2}>
                <Text style={[styles.inputTitle, {marginTop: 10, color: 'white'},]}>Question:</Text>
                <TextInput
                    testID={"QuestionTextInput"}
                    multiline
                    value={question}
                    style={[styles.inputBox, {marginBottom: 10}]}
                    returnkeyType='Done'
                    onChangeText={(val) => setQuestion(val)}
                />
                <View style={{flexDirection: 'row', alignItems: 'center'}}>
                    <View style={{flex: 1, height: 3, backgroundColor: 'white'}}/>
                </View>
                <Text style={[styles.inputTitle, {color: 'white'}]}>Answer:</Text>
                <TextInput
                    testID={"AnswerTextInput"}
                    multiline
                    value={answer}
                    style={[styles.inputBox, {marginBottom: 10}]}
                    returnkeyType='Done'
                    onChangeText={(val) => setAnswer(val)}
                />
            </View>
            {/*</LinearGradient>>*/}
            <Button
                style={stylejs.defaultButton}
                onPress={() => {
                    createQuestion().then(() => console.log("Question successfully created"));
                }}>
                <Text style={[stylejs.modal_text, {color: "white"}]}>Create</Text>
            </Button>
        </SafeAreaView>
    );
}

const styles = StyleSheet.create({
    container: {
        justifyContent: 'space-between',
        alignItems: 'center',
        top:30,
    },
    inputContainer: {
        width: '95%',
        height: '65%',
        justifyContent: 'space-around',
        alignItems: 'center',
        borderRadius: 20,
        marginBottom: 15,        
    },
    inputContainer2: {
        width: '95%',
        height: '65%',
        justifyContent: 'space-around',
        alignItems: 'center',
        borderRadius: 20,
        marginBottom: 15,
        backgroundColor:'blue'
    },
    inputBox: {
        borderRadius: 20,
        width: '90%',
        height: '30%',
        color: 'white',
    },
    inputTitle: {
        fontSize: 18,
        alignSelf: 'flex-start',
        marginLeft: 15,
    },
});

export default CreateCard
