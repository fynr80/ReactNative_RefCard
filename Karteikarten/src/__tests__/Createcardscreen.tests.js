import React from 'react';
import {NavigationContainer} from "@react-navigation/native";
import {fireEvent, render} from "@testing-library/react-native";
import CreateCard from "../screens/CreateCard";
import firebaseConfig from "../firebaseConfig";


describe("Createcard Component", () => {

    it('renders correctly', async () => {
        const navigate = jest.fn();
        const route = {params: { id: "randomID", categoriesName: "MyCategory", categoryColor: '#FD4365'}}
        const tree = render(
            <NavigationContainer>
                <CreateCard route={route} navigation={{ navigate }}/>
            </NavigationContainer>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('text change behaviour should work correctly', async () => {
        const navigate = jest.fn();
        const route = {params: { id: "randomID", catId: "randomCatId", titel: "randomTitle", categoryColor: '#FD4365'}}
        const {getByTestId} = render(
            <NavigationContainer>
                <CreateCard route={route} navigation={{ navigate }}/>
            </NavigationContainer>
        )

        const questionInput = getByTestId("QuestionTextInput");
        const answerInput = getByTestId("AnswerTextInput");

        // Text should be empty at the beginning
        expect(questionInput.props.value).toEqual("");
        expect(answerInput.props.value).toEqual("");

        // Change values of the Input
        fireEvent.changeText(questionInput, "Question 1");
        fireEvent.changeText(answerInput, "Answer 1");

        expect(questionInput.props.value).toEqual("Question 1");
        expect(answerInput.props.value).toEqual("Answer 1");

    });

})
