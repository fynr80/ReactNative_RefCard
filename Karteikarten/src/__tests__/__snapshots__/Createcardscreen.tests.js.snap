// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Createcard Component renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "space-between",
      "top": 30,
    }
  }
>
  <RNCSafeAreaView
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingLeft": 2,
          "paddingRight": 2,
        },
        Object {
          "alignItems": "flex-start",
          "justifyContent": "center",
          "top": -50,
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      CREATE CARDS
    </Text>
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RNCSafeAreaView>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        },
        Object {
          "flex": 1,
          "textAlign": "center",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderRadius": 20,
        "height": "65%",
        "justifyContent": "space-around",
        "marginBottom": 15,
        "width": "95%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "fontSize": 18,
            "marginLeft": 15,
          },
          Object {
            "color": "white",
            "marginTop": 10,
          },
        ]
      }
    >
      Question:
    </Text>
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      returnkeyType="Done"
      style={
        Array [
          Object {
            "borderRadius": 20,
            "color": "white",
            "height": "30%",
            "width": "90%",
          },
          Object {
            "marginBottom": 10,
          },
        ]
      }
      testID="QuestionTextInput"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "height": 3,
          }
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "fontSize": 18,
            "marginLeft": 15,
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      Answer:
    </Text>
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      returnkeyType="Done"
      style={
        Array [
          Object {
            "borderRadius": 20,
            "color": "white",
            "height": "30%",
            "width": "90%",
          },
          Object {
            "marginBottom": 10,
          },
        ]
      }
      testID="AnswerTextInput"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FD4365",
        "borderColor": "transparent",
        "borderRadius": 15,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "justifyContent": "center",
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 15,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 24,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Create
          </Text>
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
